tests/dotnet/UnitTests/ResourcePrefixTest.cs

```diff
@@ -0,0 +1,160 @@
+		}
+
+		[Test]
+		public void ResourcePrefix_Precedence()
+		{
+			// Arrange
+			var appBundlePrefix = "AppPrefix";
+			var platformPrefix = "PlatformPrefix";
+			
+			// Act & Assert
+			
+			// iOS - AppBundleResourcePrefix should take precedence over IPhoneResourcePrefix
+			var iOSValue = GetResourcePrefix("ios", 
+				("AppBundleResourcePrefix", appBundlePrefix), 
+				("IPhoneResourcePrefix", platformPrefix));
+			Assert.AreEqual(appBundlePrefix, iOSValue, "iOS should prioritize AppBundleResourcePrefix over IPhoneResourcePrefix");
+			
+			// tvOS - AppBundleResourcePrefix should take precedence over IPhoneResourcePrefix
+			var tvOSValue = GetResourcePrefix("tvos", 
+				("AppBundleResourcePrefix", appBundlePrefix), 
+				("IPhoneResourcePrefix", platformPrefix));
+			Assert.AreEqual(appBundlePrefix, tvOSValue, "tvOS should prioritize AppBundleResourcePrefix over IPhoneResourcePrefix");
+			
+			// Mac Catalyst - AppBundleResourcePrefix should take precedence over IPhoneResourcePrefix
+			var macCatalystValue = GetResourcePrefix("maccatalyst", 
+				("AppBundleResourcePrefix", appBundlePrefix), 
+				("IPhoneResourcePrefix", platformPrefix));
+			Assert.AreEqual(appBundlePrefix, macCatalystValue, "Mac Catalyst should prioritize AppBundleResourcePrefix over IPhoneResourcePrefix");
+			
+			// macOS - AppBundleResourcePrefix should take precedence over XamMacResourcePrefix
+			var macOSXamValue = GetResourcePrefix("macos", 
+				("AppBundleResourcePrefix", appBundlePrefix), 
+				("XamMacResourcePrefix", platformPrefix));
+			Assert.AreEqual(appBundlePrefix, macOSXamValue, "macOS should prioritize AppBundleResourcePrefix over XamMacResourcePrefix");
+			
+			// macOS - AppBundleResourcePrefix should take precedence over MonoMacResourcePrefix
+			var macOSMonoValue = GetResourcePrefix("macos", 
+				("AppBundleResourcePrefix", appBundlePrefix), 
+				("MonoMacResourcePrefix", platformPrefix));
+			Assert.AreEqual(appBundlePrefix, macOSMonoValue, "macOS should prioritize AppBundleResourcePrefix over MonoMacResourcePrefix");
+		}
+
+		private string GetResourcePrefix(string platform, params (string Property, string Value)[] properties)
+		{
+			// Create a temporary test project
+			var testDirectory = CreateTemporaryTestDirectory();
+			var projectPath = Path.Combine(testDirectory, "TestApp.csproj");
+			
+			// Create project file with specified properties
+			File.WriteAllText(projectPath, GetTestProjectContent(platform, properties));
+			
+			// Use dotnet build with getProperty to get _ResourcePrefix value
+			var (exitCode, stdout, stderr) = ExecuteGetPropertyCommand(projectPath);
+			Assert.Zero(exitCode, $"dotnet build failed: {stderr}");
+			
+			// Return the property value from stdout
+			return stdout.Trim();
+		}
+
+		private (int ExitCode, string StdOut, string StdErr) ExecuteGetPropertyCommand(string projectPath)
+		{
+			return ExecutionHelper.RunCommand(
```

Comment:
Add a new method called `GetProperty` in tests/common/DotNet.cs that will run dotnet to get the property.